sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -37,6 +39,26 @@ import org.apache.spark.sql.types.StructType
 class JoinSuite extends QueryTest with SharedSparkSession {
   import testImplicits._
 
+  private def attachCleanupResourceChecker(plan: SparkPlan): Unit = {
+    // SPARK-21492: Check cleanupResources are finally triggered in SortExec node for every
+    // test case
+    val sorts = new ArrayBuffer[SortExec]()
```

Comment:
super nit: now we don't need this array
```
plan.foreachUp {
  case s: SortExec =>
    // add spy
  case _ =>
}
```

Reply:
Yep, just do the simplify at the same time :) 6d6dd5a